pypxe/tftp.py

```diff
@@ -121,39 +147,61 @@ def read(self, address, message):
             response += str(filesize)
             response += chr(0)
 
-        socknew = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-        socknew.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-        socknew.bind((self.ip, 0))
-        self.ongoing[address]['sock'] = socknew
+        # Create the data socket only if needed
+        if self.ongoing[address]['sock'] == None:
+            socknew = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+            socknew.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+            socknew.bind((self.ip, 0))
+            self.ongoing[address]['sock'] = socknew
 
         if response:
             response = struct.pack('!H', 6) + response
             socknew.sendto(response, address)
-        else:
+        # if no options in request and no block sent yet start sending data
+        elif self.ongoing[address]['block'] == 1:
             self.sendBlock(address)
+        else:
+            self.logger.warning('Ignored TFTP request: no options in a middle of a transmission')
```

Comment:
I don't think there is an error we can send. However, it would require either a very naughty client, or a very broken network. We only increment `block` when we get an ack, so the dupe read would have to come after a block 1 ack. I think this would be so wrong it's not our problem.